src/node_file.cc

```diff
@@ -992,6 +993,28 @@ void Close(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void FastClose(Local<Object> recv,
+                      const int32_t fd,
+                      // NOLINTNEXTLINE(runtime/references) This is V8 api.
+                      v8::FastApiCallbackOptions& options) {
+  Environment* env = Environment::GetCurrent(recv->GetCreationContextChecked());
+
+  uv_fs_t req;
+  FS_SYNC_TRACE_BEGIN(close);
+  int err = uv_fs_close(nullptr, &req, fd, nullptr) < 0;
+  FS_SYNC_TRACE_END(close);
+  uv_fs_req_cleanup(&req);
+
+  if (err < 0) {
+    options.fallback = true;
+  } else {
+    // Only remove unmanaged fds if the close was successful.
+    env->RemoveUnmanagedFd(fd);
```

Comment:
This is unsafe, `RemoveUnmanagedFd()` can call `ProcessEmitWarning()` which calls back into JS `process.emitWarning()`. Although since it's just a warning, you could probably just add a parameter to `RemoveUnmanagedFd` to make it invoke `ProcessEmitWarning()` later in a native Immediate.

Reply:
If I understood you correctly, inside `RemoveUnmanagedFd` you want me to conditionally call the following right?

```
SetImmediateThreadsafe([](Environment* env) {
  ProcessEmitWarning(
        this, "File descriptor %d closed but not opened in unmanaged mode", fd);
});
```

If not, what is the correct way to schedule a native immediate?

Reply:
I think I got it. Can you review it again? @joyeecheung